### DISCUSSION TITLE
pyuvm build-upon cocotb is now available

### DISCUSSION THREAD

**Discussion by raysalemi** [cocotb/cocotb#2662] [Cocotb In The Wild]
Hi All

I have created pyuvm 2.0 which gets rid of all the threads and replaces them with coroutines as @ktbarrett suggested.  Thanks, Kaleb it is *so* much better and easier to use.

**pyuvm** now lists **cocotb** as a requirement and will install **cocotb** if it isn't installed already with this command:

     % pip install pyuvm==2.0a8

The new version is currently in beta. I'm going to write some testbenches in it as a test, and release it before DAC, where I will be giving a presentation.

Thanks to @eric-wieser , @ktbarrett , and @marlonjames for their patience in helping me debug my blocking issue.

**Comment by cmarqu** [comment]
@raysalemi We now have the `Framework :: Cocotb` classifier that you could add to https://github.com/pyuvm/pyuvm/blob/master/setup.cfg#L12

You may also want to add `Topic :: Scientific/Engineering :: Electronic Design Automation (EDA)`.

**Reply by raysalemi** [reply]
So like this?

```
classifiers =
    Programming Language :: Python :: 3
    License :: OSI Approved :: Apache Software License
    Operating System :: OS Independent
    Framework :: Cocotb
    Topic :: Scientific/Engineering :: Electronic Design Automation (EDA)
```

**Reply by cmarqu** [reply]
Yes, looks good to me.

**Reply by raysalemi** [reply]
Looks like they don't have a Cocotb Framework yet.

**Reply by raysalemi** [reply]
```
HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/
Invalid value for classifiers. Error: Classifier 'Framework :: Cocotb' is not a valid classifier.
```

**Reply by eric-wieser** [reply]
Lowercase `cocotb` not `Cocotb`

**Reply by raysalemi** [reply]
<img width="416" alt="image" src="https://user-images.githubusercontent.com/3169242/129423299-f9b6311e-40e1-4667-9df2-4850fb56dd50.png">
I don't think it is up yet. It was not in the Framework list.

**Reply by eric-wieser** [reply]
Did the error message go away when you switched to lower case?

**Comment by raysalemi** [comment]
Thanks!

**Comment by raysalemi** [comment]
I'll give it a try.
